sources/web/datalab/polymer/modules/ApiManager.ts

```diff
@@ -149,29 +148,101 @@ class ApiManager {
   }
 
   /**
+   * Create a new notebook or directory.
+   * @param type string type of the created item, can be 'notebook' or 'directory'
+   */
+  static createNewItem(type: string) {
+    return new Promise((resolve, reject) => {
+      ApiManager._xhr(ApiManager.contentApiUrl,
+          (request: XMLHttpRequest) => {
```

Comment:
Now that you are using promises, consider making _xhr return a Promise, then you can use then and catch instead of the main callback and errorCallback.